ethereum/eth/src/test/java/org/hyperledger/besu/ethereum/eth/LegacyForkIdManager.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.eth;
+
+import org.hyperledger.besu.ethereum.chain.Blockchain;
+import org.hyperledger.besu.ethereum.core.Hash;
+import org.hyperledger.besu.ethereum.eth.manager.ForkId;
+import org.hyperledger.besu.ethereum.rlp.RLPInput;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.zip.CRC32;
+
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+
+public class LegacyForkIdManager {
+
+  private final Blockchain blockchain;
+  private final Hash genesisHash;
+  private final List<Long> forks;
+  private long forkNext;
+  private final long highestKnownFork;
+  private List<ForkId> forkAndHashList;
+
+  public LegacyForkIdManager(final Blockchain blockchain, final List<Long> forks) {
+    this.blockchain = blockchain;
+    this.genesisHash = blockchain.getGenesisBlock().getHash();
+    // de-dupe and sanitize forks
+    this.forks =
+        forks.stream().filter(fork -> fork > 0).distinct().collect(Collectors.toUnmodifiableList());
+    highestKnownFork = forks.size() > 0 ? forks.get(forks.size() - 1) : 0L;
+    createForkIds();
+  };
+
+  public List<ForkId> getForkAndHashList() {
+    return this.forkAndHashList;
+  }
+
+  public ForkId getLatestForkId() {
+    if (forkAndHashList.size() > 0) {
+      return forkAndHashList.get(forkAndHashList.size() - 1);
+    }
+    return null;
+  }
+
+  public static ForkId readFrom(final RLPInput in) {
+    in.enterList();
+    final Bytes hash = in.readBytes();
+    final Bytes next = in.readBytes();
+    in.leaveList();
+    return new ForkId(hash, next);
+  }
+
+  /**
+   * EIP-2124 behaviour
+   *
+   * @param forkId to be validated.
+   * @return boolean (peer valid (true) or invalid (false))
+   */
+  boolean peerCheck(final ForkId forkId) {
```

Comment:
Do we need to keep these unused methods around?

Reply:
No. I removed them.